tests/Microsoft.Identity.Test.Integration.netcore/HeadlessTests/FmiIntegrationTests.cs

```diff
@@ -0,0 +1,150 @@
+            await RunFicFlow(fmiCredential, _fmiExchangeScope, _fmiNonRmaClientIdInpersonation).ConfigureAwait(false);
+        }
+
+        private async Task<string> RunRmaFlow(string clientId, string authority, string scope)
+        {
+            X509Certificate2 cert = CertificateHelper.FindCertificateByName(TestConstants.AutomationTestCertName);
+
+            var confidentialApp = ConfidentialClientApplicationBuilder
+                        .Create(clientId)
+                        .WithAuthority(authority, true)
+                        .WithExtraQueryParameters("dc=ESTS-PUB-SCUS-LZ1-FD000-TEST1")
+                        .WithExperimentalFeatures(true)
+                        .WithCertificate(cert, sendX5C: true)
+                        .BuildConcrete();
+
+            var appCacheRecorder = confidentialApp.AppTokenCache.RecordAccess();
+
+            var authResult = await confidentialApp.AcquireTokenForClient(new[] { scope })
+                                                    .WithFmiPath(_fmiPath)
+                                                    .ExecuteAsync()
+                                                    .ConfigureAwait(false);
+
+            MsalAssert.AssertAuthResult(authResult);
+            appCacheRecorder.AssertAccessCounts(1, 1);
+            Assert.AreEqual(TokenSource.IdentityProvider, authResult.AuthenticationResultMetadata.TokenSource);
+            Assert.IsTrue(appCacheRecorder.LastAfterAccessNotificationArgs.IsApplicationCache);
+            Assert.IsTrue(appCacheRecorder.LastAfterAccessNotificationArgs.HasTokens);
+            CollectionAssert.AreEquivalent(new[] { scope }, appCacheRecorder.LastBeforeAccessNotificationArgs.RequestScopes.ToArray());
+            CollectionAssert.AreEquivalent(new[] { scope }, appCacheRecorder.LastAfterAccessNotificationArgs.RequestScopes.ToArray());
+            Assert.AreEqual(_fmiTenantId, appCacheRecorder.LastBeforeAccessNotificationArgs.RequestTenantId ?? "");
+            Assert.AreEqual(_fmiTenantId, appCacheRecorder.LastAfterAccessNotificationArgs.RequestTenantId ?? "");
+            Assert.IsTrue(authResult.AuthenticationResultMetadata.DurationTotalInMs > 0);
+            Assert.IsTrue(authResult.AuthenticationResultMetadata.DurationInHttpInMs > 0);
+
+            return authResult.AccessToken;
+        }
+
+        private async Task RunFicFlow(string fmiCredential, string scope, string clientId)
+        {
+            var confidentialApp = ConfidentialClientApplicationBuilder
+                        .Create(clientId)
+                        .WithAuthority(_fmiAuthority, true)
+                        .WithExtraQueryParameters("dc=ESTS-PUB-SCUS-LZ1-FD000-TEST1")
+                        .WithExperimentalFeatures(true)
+                        .WithClientAssertion((options) =>
+                            {
+                                Assert.AreEqual(clientId, options.ClientID);
+                                return Task.FromResult(fmiCredential);
```

Comment:
This is a bad way of using a FIC. If ppl use this code in production, it will lead to fmiCredential going stale (expiring) and cause outages. 

Please use the recommended pattern